plugins/uefi/fu-plugin-uefi.c

```diff
@@ -91,6 +91,81 @@ fu_plugin_get_results (FuPlugin *plugin, FuDevice *device, GError **error)
 	return TRUE;
 }
 
+static void
+fu_plugin_uefi_add_hsi_tainted (FuPlugin *plugin, GPtrArray *attrs)
+{
+	FwupdSecurityAttr *attr = fwupd_security_attr_new ("org.kernel.CheckTainted");
+	fwupd_security_attr_set_name (attr, "Linux Kernel");
+	fwupd_security_attr_add_flag (attr, FWUPD_SECURITY_ATTR_FLAG_RUNTIME_ISSUE);
+	if (fu_uefi_read_file_as_uint64 ("/proc/sys/kernel", "tainted") == 0) {
+		fwupd_security_attr_add_flag (attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS);
+	} else {
+		fwupd_security_attr_set_summary (attr, "Tainted");
+	}
+	g_ptr_array_add (attrs, attr);
+}
+
+static void
+fu_plugin_uefi_add_hsi_lockdown (FuPlugin *plugin, GPtrArray *attrs)
+{
+	FwupdSecurityAttr *attr = fwupd_security_attr_new ("org.kernel.CheckLockdown");
+	gsize bufsz = 0;
+	g_autofree gchar *buf = NULL;
+	fwupd_security_attr_set_name (attr, "Linux Kernel");
+	fwupd_security_attr_add_flag (attr, FWUPD_SECURITY_ATTR_FLAG_RUNTIME_ISSUE);
+	if (g_file_get_contents ("/sys/kernel/security/lockdown", &buf, &bufsz, NULL)) {
+		if (g_strstr_len (buf, bufsz, "[integrity]") != NULL ||
+		    g_strstr_len (buf, bufsz, "[confidentiality]") != NULL) {
+			fwupd_security_attr_add_flag (attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS);
+			fwupd_security_attr_set_summary (attr, "Locked down");
+		}
+	}
```

Comment:
I don't think either taint or lockdown should be specific to UEFI architecture.

Reply:
A new "linux" plugin perhaps?

Reply:
Ya that sounds good

Reply:
That or merging swap functionality into a Linux plug-in

Reply:
> merging swap functionality into a Linux plug-in

It's certainly easier disabling plugins when the plugins just do one thing. I'd err on more simple plugins from a readability and maintenance point of view.

Reply:
OK.

Reply:
I've split these out; I think I was right on lots of small plugins, there are lots of edge cases (e.g. like /sys/kernel/tainted) changing that we should handle.